source/iface/device-selector.rst

```diff
@@ -3,116 +3,94 @@
+selects the one which returns the highest score. If the highest
+value is strictly negative no device is selected.
 
-  sycl::device select_device() const;
+In places where only one device has to be picked and the high score is
+obtained by more than one device, then one of the tied devices will be
+returned, but which one is not defined and may depend on enumeration
+order, for example, outside the control of the SYCL runtime.
 
 
-Returns the device with the highest score as determined by calling
-`operator()`_.
-
-.. rubric:: Exceptions
-
-Throws a runtime error if all devices have a negative score.
-
-``operator=``
-=============
-
-::
-
-  sycl::device_selector &operator=(const sycl::device_selector &rhs);
+.. _built-in-device-selectors:
 
-Create a device selector by copying another one.
+===========================
+Built-in Device Selectors
+===========================
 
+Standard device selectors included with all SYCL implementations:
 
-``operator()``
-==============
+.. list-table::
 
-::
+  * - ``default_selector_v``
+    - Select a SYCL device from any supported SYCL backend based on an implementation-defined
+      heuristic. Must select the host device if no other suitable device can be found.
+  * - ``gpu_selector_v``
+    - Select a SYCL device from any supported SYCL backend for which the device type is ``info::device::device_type::gpu``.
+  * - ``accelerator_selector_v``
+    - Select a SYCL device from any supported SYCL backend for which the device type is ``info::device_type::accelerator``.
+  * - ``cpu_selector_v``
+    - Select a SYCL device from any supported SYCL backend for which the device type is ``info::device_type::cpu``.
+  * - ::
 
-  virtual int operator()(const sycl::device &device) const = 0;
+        __unspecified_callable__
+        aspect_selector(const std::vector<aspect>& aspectList,
+                        const std::vector<aspect>& denyList = {});
 
-Scoring function for devices.
+        template <typename... AspectList>
+        __unspecified_callable__ aspect_selector(AspectList... aspectList);
 
-All derived device selectors must define this
-operator. `select_device` calls this operator for every device, and
-selects the device with highest score. Return a negative score if a
-device should not be selected.
+        template <aspect... AspectList>
+        __unspecified_callable__ aspect_selector();
 
+    - The free function ``aspect_selector`` has several overloads, each of which returns a selector object that selects a SYCL
+      device from any supported SYCL backend which contains all the requested aspects.
+      If no aspects are passed in, the generated selector behaves like ``default_selector``.
 
-.. _built-in-device-selectors:
+      Required aspects can be passed in as a vector, as function arguments, or as template parameters, depending on the
+      function overload. The function overload that takes aspectList as a vector takes another vector argument denyList
```

Comment:
```suggestion
      function overload. The function overload that takes ```aspectList``` as a vector takes another vector argument ```denyList```
```

Would it be better to use code font for parameter names?  I saw that you needed to add "aspectList" and "denyList" to the "spelling_wordlist.txt" file.  If these words were in code font, would the tools automatically ignore their spelling?

Reply:
Yes, added code font and removed these words from the ignore list. They are automatically ignored

Reply:
@rscohn2 put this in the [README](https://github.com/KhronosGroup/SYCL_Reference/blob/main/README.rst):

> If the spell checker is warning about code, it means you have not properly delimited the code in the rst. Do not add it to spelling_wordlist.txt add, back quotes or other delimeters so it will be ignored.

So I think that making these into code font aligns with that direction.